src/pyobo/sources/gtdb.py

```diff
@@ -0,0 +1,111 @@
+    "f__": Reference(prefix="TAXRANK", identifier="0000004", name="family"),
+    "g__": Reference(prefix="TAXRANK", identifier="0000005", name="genus"),
+    "s__": Reference(prefix="TAXRANK", identifier="0000006", name="species"),
+}
+
+GTDB_AR_URL = "https://data.gtdb.ecogenomic.org/releases/latest/ar53_metadata.tsv.gz"
+GTDB_BAC_URL = "https://data.gtdb.ecogenomic.org/releases/latest/bac120_metadata.tsv.gz"
+
+logger = logging.getLogger(__name__)
+
+
+class GTDBGetter(Obo):
+    """An ontology representation of the GTDB taxonomy."""
+
+    ontology = bioversions_key = PREFIX
+    typedefs = [has_taxonomy_rank]
+
+    def iter_terms(self, force: bool = False) -> Iterable[Term]:
+        """Iterate over terms in the ontology."""
+        return iter_terms(version=self._version_or_raise, force=force)
+
+
+def _parse_gtdb_taxonomy(tax_string: str) -> list[tuple[str, str]]:
+    """Parse GTDB taxonomy string into (level, name) tuples."""
+    taxa = []
+    parts = [p.strip() for p in tax_string.split(";") if p.strip()]
+
+    for part in parts:
+        if len(part) < 4 or "__" not in part:
+            logger.warning(f"Malformed taxon string: {part}")
+            continue
+        level = part[:3]
+        name = part[3:].strip()
+        if level in LEVEL_ORDER and name:
+            taxa.append((level, name))
+        else:
+            logger.warning(f"Invalid taxonomic level or name in part: {part}")
+    return taxa
+
+
+def iter_terms(version: str, force: bool = False) -> Iterable[Term]:
+    """Iterate over GTDB terms."""
+    ar_path = ensure_path(PREFIX, url=GTDB_AR_URL, version=version, force=force)
+    bac_path = ensure_path(PREFIX, url=GTDB_BAC_URL, version=version, force=force)
+    columns = ["gtdb_taxonomy", "ncbi_taxid"]
+    for path_name, path in [
+        ("ar", ar_path),
+        ("bac", bac_path),
+    ]:
+        df = pd.read_csv(path, sep="\t", dtype=str)
+        for tax_string, ncbitaxon_id in tqdm(
+            df[columns].values, desc=f"[{PREFIX}] processing {path_name}", unit_scale=True
+        ):
+            yield from _process_row(tax_string, ncbitaxon_id)
+
+
+def _process_row(tax_string, ncbitaxon_id) -> Iterable[Term]:
+    if not isinstance(tax_string, str):
+        logger.warning(f"Invalid taxonomy string: {tax_string}")
+        return None
+
+    taxa = _parse_gtdb_taxonomy(tax_string)
+    if not taxa:
+        logger.warning(f"No valid taxa found in: {tax_string}")
+        return None
+
+    parent_reference = None
+    for level, name in taxa:
+        identifier = f"{level}{name.replace(' ', '_')}"
+        term = Term(
+            reference=Reference(prefix=PREFIX, identifier=identifier, name=name),
+        )
+        if taxrank_reference := LEVEL_TO_TAXRANK.get(level):
+            term.append_property(has_taxonomy_rank, taxrank_reference)
+
+        if parent_reference:
+            term.append_parent(parent_reference)
+        if ncbitaxon_id and level == "s__":
```

Comment:
It is important to note that there is 1 to many relantionship between GTDB to NCBI, at the extreme you will see the term "s__Staphylococcus aureus" has 4202 different ncbitax ids annotated as dbxref. 
GTDB does not provide mappings to other ncbi taxonomy levels rather than species, I message them asking about it and this is what I heard 
"We do not provide a mapping between GTDB taxa and NCBI Taxon IDs. In general, the mapping may be many-to-many so this is not straight forward and in some cases taxa defined in these two taxonomy have different circumscriptions (definitions) (e.g.: [https://www.biorxiv.org/content/10.1101/2021.09.22.461432v1](https://urldefense.us/v3/__https://www.biorxiv.org/content/10.1101/2021.09.22.461432v1__;!!G_uCfscf7eWS!eFb3uhYz-BUlLQdQHqgXH0_hnZpBtNK8Vld8Xf6a1WY7a6loPMu1NzhRhKPuA4s5LOK83agTpoMDWcV6b0WVhsI$)). We do provide files that show how NCBI taxa relate to GTDB taxa or vice versa in the [https://data.gtdb.ecogenomic.org/releases/latest/auxillary_files](https://urldefense.us/v3/__https://data.gtdb.ecogenomic.org/releases/latest/auxillary_files__;!!G_uCfscf7eWS!eFb3uhYz-BUlLQdQHqgXH0_hnZpBtNK8Vld8Xf6a1WY7a6loPMu1NzhRhKPuA4s5LOK83agTpoMDWcV6IZ1kEEU$) directory."

Reply:
When you look at the auxilliary files, still no clear better mappings than what we already have.

Reply:
hmm okay then these seem more like "see also" rather than a dbxref since it's not clear what is is the context of the association. you can take a look at SSSOM (https://mapping-commons.github.io/sssom/) for a bit more information on how to do more precise mappings - the idea is to try and avoid "dbxrefs" since they don't really tell you anything